src/types/state.ts

```diff
@@ -0,0 +1,25 @@
+import { ExtractParamTypes } from '@/types/params'
+import { Param } from '@/types/paramTypes'
+import { Routes } from '@/types/route'
+import { RouteGetByKey } from '@/types/routeWithParams'
+
+export type RouteState<T extends Record<string, unknown>> = {
+  scrollPosition?: number,
+  pageTitle?: string,
```

Comment:
Agreed these will likely exist at some point. But we should probably leave them out for now. 
```suggestion
```